app/main.py

```diff
@@ -39,51 +48,257 @@ def execute(self):
 
             self.execute_as_loop(30)  # 30-second interval.
         """
+        self.load_shelve()
 
     def shutdown(self):
         """Run when your NApp is unloaded.
 
         If you have some cleanup procedure, insert it here.
         """
 
+    @listen_to("kytos/topology.unloaded")
+    def unload_topology(self):  # pylint: disable=W0613
+        """Function meant for validation, to make sure that the shelve
+        has been loaded before all the other functions that use it begins to
+        call it."""
+        self.shelve_loaded = False  # pylint: disable=W0201
+
+    def test_kytos_topology(self):
+        """ Test if the Topology napp has loaded """
+        try:
+            _ = self.get_kytos_topology()
+            return True
+        except Exception as err:  # pylint: disable=W0703
+            log.debug(err)
+            return False
+
+    @staticmethod
+    def get_kytos_topology():
+        """retrieve topology from API"""
+        kytos_topology = requests.get(
+                settings.KYTOS_TOPOLOGY, timeout=10).json()
+        return kytos_topology["topology"]
+
+    def validate_sdx_topology(self):
+        """ return 200 if validated topology following the SDX data model"""
+        try:
+            response = requests.post(
+                    settings.SDX_TOPOLOGY_VALIDATE,
+                    json=self.sdx_topology,
+                    timeout=10)
+        except ValueError as exception:
+            log.info("validate topology result %s %s", exception, 401)
+            raise HTTPException(
+                    401,
+                    detail=f"Path is not valid: {exception}"
+                ) from exception
+        return {"result": response.json(), "status_code": response.status_code}
+
+    def convert_topology(self, event_type=0, event_timestamp=None):
+        """Function that will take care of update the shelve containing
+        the version control that will be updated every time a change is
+        detected in kytos topology, and return a new sdx topology"""
+        try:
+            # open the topology shelve
+            with shelve.open("topology_shelve") as open_shelve:
+                version = open_shelve['version']
+                self.dict_shelve = dict(open_shelve)  # pylint: disable=W0201
+                open_shelve.close()
+            if version >= 1 and event_type != 0:
```

Comment:
Hi @italovalcy, we need to identify the first time we create the topology so that we can send the whole topology to sdx lc. They need to identify this as the first time they have a topology, probably with a different endpoint, so they can take the necessary steps for the first time. Afterward, we can initialize the variable with 1. If you have any suggestions how to handle this, I appreciate your help.

Reply:
we probably implement a get request from sdx-lc to do this: If the topology is available and the version is equal to 0, the counting will start at 1.

Reply:
In my opinion, the only thing that must be persistent for this napp is the version number advertised to the SDX-LC. The state and status of the network must be always "fresh".